### ISSUE TITLE
Some API functions are not handled correctly by Doxygen

### ISSUE THREAD

**Issue by kbevers** [OSGeo/PROJ#2540]
Originally discovered in https://github.com/OSGeo/PROJ/pull/2507. If the function `proj_context_set_search_paths()` is added to the Sphinx/Breathe docs using a `doxygenfunction` directive like

```
.. doxygenfunction:: proj_context_set_search_paths
   :project: doxygen_api

```
 the doc generation fails:

```
$ make html
../scripts/doxygen.sh
No Doxygen warnings found
No Doxygen warnings found
touch .doxygen_up_to_date
sphinx-build -b html -d build/doctrees   source -W build/html
Running Sphinx v3.0.3
Initializing Spelling Checker
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 239 source files that are out of date
updating environment: [new config] 239 added, 0 changed, 0 removed
checking for /Users/kevers/dev/proj/repo/docs/source/references.bib in bibtex cache... not found
parsing bibtex file /Users/kevers/dev/proj/repo/docs/source/references.bib... parsed 41 entries


Warning, treated as error:
/Users/kevers/dev/proj/repo/docs/source/development/reference/functions.rst:787:doxygenfunction: Cannot find function "proj_context_set_search_paths" in doxygen xml output for project "doxygen_api" from directory: ../build/xml/
make: *** [html] Error 2
```

Inspecting the docygen XML output reveals that the file is at least partially recognized by Doxygen

```
$ grep proj_context_set_search_paths docs/build/xml/*
docs/build/xml/filemanager_8cpp.xml:        <definition>void proj_context_set_search_paths</definition>
docs/build/xml/filemanager_8cpp.xml:        <name>proj_context_set_search_paths</name>
docs/build/xml/filemanager_8cpp.xml:<codeline lineno="1940"><highlight class="normal"></highlight><highlight class="comment">/*<sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/>proj_context_set_search_paths()<sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/><sp/>*/</highlight><highlight class="normal"></highlight></codeline>
docs/build/xml/filemanager_8cpp.xml:<codeline lineno="1960"><highlight class="keywordtype">void</highlight><highlight class="normal"><sp/>proj_context_set_search_paths(PJ_CONTEXT<sp/>*ctx,<sp/></highlight><highlight class="keywordtype">int</highlight><highlight class="normal"><sp/>count_paths,</highlight></codeline>
docs/build/xml/filemanager_8cpp.xml:<codeline lineno="1985"><highlight class="normal"><sp/><sp/><sp/><sp/>proj_context_set_search_paths(</highlight><highlight class="keyword">nullptr</highlight><highlight class="normal">,<sp/>count,</highlight></codeline>
docs/build/xml/index.xml:    <member refid="filemanager_8cpp_1afe6b85648984b25309b399c68716f914" kind="function"><name>proj_context_set_search_paths</name></member>
docs/build/xml/proj_8h.xml:<codeline lineno="370"><highlight class="normal"></highlight><highlight class="keywordtype">void</highlight><highlight class="normal"><sp/>PROJ_DLL<sp/>proj_context_set_search_paths(PJ_CONTEXT<sp/>*ctx,<sp/></highlight><highlight class="keywordtype">int</highlight><highlight class="normal"><sp/>count_paths,<sp/></highlight><highlight class="keyword">const</highlight><highlight class="normal"><sp/></highlight><highlight class="keywordtype">char</highlight><highlight class="normal">*<sp/></highlight><highlight class="keyword">const</highlight><highlight class="normal">*<sp/>paths);</highlight></codeline>
```

 however when looking for it in the raw Doxygen HTML output it is not to be found. The same goes for a couple of other functions in `src/filemanager.cpp`: `proj_context_set_file_finder()` and `proj_context_set_ca_bundle_path()`. On the other hand a function like `proj_context_set_sqlite3_vfs_name()` shows up just fine even though there are no obvious difference between the way it is documented compared to the functions that don't show up in the docs.

**Comment by direvus** [comment]
Is this one solved, now that #2944 has been merged?  Or is there more to be done?

**Comment by kbevers** [comment]
As far as I can tell from a quick inspection `proj_context_set_file_finder()` and `proj_context_set_ca_bundle_path()` still needs to be taken care of.
